src/app/directives/card-marker-color.directive.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  Directive, ElementRef, Input, OnInit, Renderer2,
+} from '@angular/core';
+
+@Directive({
+  selector: '[appCardMarkerColor]',
+})
+export class CardMarkerColorDirective implements OnInit {
+  colorClass = '';
+
+  @Input() Date?:string;
+
+  constructor(private elementref: ElementRef,
+    private renderer: Renderer2) {
+  }
+
+  ngOnInit() {
+    this.defineColor();
+    this.renderer.addClass(this.elementref.nativeElement, this.colorClass);
+  }
+
+  defineColor() {
+    const date = new Date(this.Date as string).getTime();
+    if (date > this.SevenDaysPassedTime()) {
```

Comment:
I suggest you to not use else if. Create a method or better to util function wich will have 
```
if (statement1) return 'blue';
if (statement2) return 'green'; 
...